core/common/src/main/java/alluxio/underfs/UnderFileSystemConfiguration.java

```diff
@@ -114,6 +120,19 @@ public UnderFileSystemConfiguration setShared(boolean shared) {
   public UnderFileSystemConfiguration createMountSpecificConf(Map<String, String> mountConf) {
     UnderFileSystemConfiguration ufsConf = new UnderFileSystemConfiguration(mProperties.copy());
     ufsConf.mProperties.merge(mountConf, Source.MOUNT_OPTION);
+    String ufsConfigValue = ufsConf.get(PropertyKey.UNDERFS_HDFS_CONFIGURATION);
+    if (ufsConfigValue.startsWith(Name.UNDERFS_HDFS_CONFIGURATION_PREFIX)) {
+      Configuration hdfsConf = new Configuration();
+      for (String path :
+          ufsConfigValue.substring(Name.UNDERFS_HDFS_CONFIGURATION_PREFIX.length()).split(":")) {
+        if (!path.isEmpty()) {
+          hdfsConf.addResource(new Path(path));
+        }
+      }
+      ufsConf.merge(hdfsConf.getPropsWithPrefix(Name.UNDERFS_HDFS_CONFIGURATION_PREFIX),
```

Comment:
I don't know how this works.

 I thought this would convert the hdfs conf to key/values. However, this is taking all the property keys which have the prefix of `UNDERFS_HDFS_CONFIGURATION_PREFIX`, which is "kv@". No hdfs properties will have that prefix, right?

Reply:
Oh, sorry, I make a mistake here, I suppose to `getPropsWithPrefix("")` to merge all hdfs configuration from the given files. Thank you for point out this bug.